### PR TITLE
CI: Add in deploy tooling

### PR DESCRIPTION
**Pull request by antdking**
TODO: update reports to perform conditional upload

- Using a Makefile as it does recipes based on globs better than other tooling (we need to vary based on ipynb vs py)
- rules for happy-path laid out in `CONTRIBUTING.md`

### CODE REVIEW

**Comment by antdking** [comment]
Lost focus a little yesterday.
Fly deploys are now in a python script; we can move to CloudApps when ready to streamline the process.

Some apps have been excluded from deploying:
- background-remover does too much processing on boot, so never comes alive in time
- some apps depend on dp-components. They'll go live later

**Comment by mands** [comment]
Ok sounds great,

- `background-remover` - I can perhaps rewrite this later to do that processing on first request or in the background later on next week

**Review by mands — APPROVED** [review]
This looks great - thanks for creating! Happy to merge when you're ready

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+If using a Notebook (`.ipynb`), avoid use of Magics (`%` or `!` statements)
+
+#### Supporting upload and save
+
+When deploying, an environment variable `DATAPANE_DEPLOY=1` is set.
+This can be used to select the action a report takes:
+
+```py
+import os
+if os.environ.get('DATAPANE_DEPLOY') == "1":
```
Comment (mands) [inline CONTRIBUTING.md:47]: cool, so I can just go into each report and add this snippet for the ones we want to upload?

Comment (antdking) [reply CONTRIBUTING.md:47]: Pretty much, yeah

**Comment by antdking** [comment]
Merging this as-is.
As with the other PR, the secrets aren't configured yet, so it won't do anything

**Comment by antdking** [comment]
@mands can the dev group get rights to merge please

**Pull request merged**